### PR TITLE
[ML][Transforms] add CCS capabilities to pivot transforms

### PR DESCRIPTION
**Pull request by benwtrent**
This is the first step for CCS support in transforms.

There is one outstanding issue in regards to this:

- How do we generate checkpoints for remote indices?

From what I can tell, the only way to get cluster aliases is via the `TransportService`. This service does not seem to be available outside of transport classes, at least it is not available via the `createComponents` logic in the `DataFrame.java` class. 

It seems to me that the only way to generate checkpoints for remote indices is to add a `remote_clusters` field to the transform configuration. This field could get set on `PUT` and `UPDATE`. This will allow the transform to pass in the known remote cluster aliases (given the remote indices configured) to the checkpoint service.

In short, the checkpoint service needs to know the cluster aliases to create a remote capable client.

Opening as WIP to keep track of work. Given other priorities, this may not be merged for some time.

### CODE REVIEW

**Comment by hendrikmuhs** [comment]
@benwtrent can be closed?

(It was a great helper BTW)
